mllib/src/main/scala/org/apache/spark/ml/classification/Classifier.scala

```diff
@@ -179,7 +230,7 @@ abstract class ClassificationModel[FeaturesType, M <: ClassificationModel[Featur
         }
         predictUDF(col(getFeaturesCol))
       }
-      outputData = outputData.withColumn(getPredictionCol, predUDF)
+      outputData = outputData.withColumn(getPredictionCol, predUDF, predictionMetadata)
```

Comment:
The type of ```predictionMetadata``` is ```Metadata``` rather than ```Attribute```, so we can not use ```withName``` method. Actually ```predictionMetadata``` has been set name with ```getPredictionCol``` in ```generatePredictionMetadata```.